engines/nancy/state/loadsave.cpp

```diff
@@ -99,7 +103,61 @@ void LoadSaveMenu::process() {
 	g_nancy->_cursor->setCursorType(CursorManager::kNormalArrow);
 }
 
+void LoadSaveMenu::scummVMSave() {
+	GUI::SaveLoadChooser *dialog = new GUI::SaveLoadChooser(_("Save game:"), _("Save"), true);
+	int slot = dialog->runModalWithCurrentTarget();
+	Common::String saveName = dialog->getResultString();
+	delete dialog;
+
+	g_nancy->_graphics->suppressNextDraw();
+	_destroyOnExit = true;
+	_state = kStop;
+	_selectedSave = slot;
+
+	if (slot >= 0) {
+		g_nancy->saveGameState(slot, saveName, false);
+		g_nancy->_hasJustSaved = true;
+	}
+}
+
+void LoadSaveMenu::scummVMLoad() {
+	GUI::SaveLoadChooser *dialog = new GUI::SaveLoadChooser(_("Load game:"), _("Load"), false);
+	int slot = dialog->runModalWithCurrentTarget();
+	delete dialog;
+
+	g_nancy->_graphics->suppressNextDraw();
+	_destroyOnExit = true;
+	_state = kStop;
+	_selectedSave = slot;
+
+	if (slot >= 0) {
+		if (Nancy::State::Scene::hasInstance())
+			Nancy::State::Scene::destroy();
+
+		ConfMan.setInt("save_slot", slot, Common::ConfigManager::kTransientDomain);
+
+		_enteringNewState = true;
+	}
+}
+
 void LoadSaveMenu::onStateEnter(const NancyState::NancyState prevState) {
+
+	if (!ConfMan.getBool("originalsaveload")) {
+		if (Nancy::State::Scene::hasInstance()) {
+			GUI::MessageDialog saveOrLoad(_("Would you like to load or save a game?"), _("Load"), _("Save"));
```

Comment:
Just noticed this - when exiting the game through the main menu, the engine shows a "Would you like to save your game before exiting?". Clicking yes still leads to this code path, and asks the player whether they want to load or save. But since they said yes to that question, they'll want to save.

To solve this, I'd add a transient ConfMan bool in savegame.cpp, stop() case 6, and then check for it (and clear it) on this line.

Reply:
Fixed

Reply:
@fracturehill Fixed in 7d2655c3c75fed2057598c7b129d5f3b2cc933b4.
I've modified the ScummVM save dialog to return to the main menu, like the original dialog does.

The original load dialog takes the user back to the scene upon load, so I've kept that behavior the same in the ScummVM load dialog.